spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -151,15 +154,21 @@ private SparkTableUtil() {
    *
    * @param spark a Spark session
    * @param tableIdent a table identifier
+   * @param partitionFilter the partition filter
    * @return all table's partitions
    */
-  public static List<SparkPartition> getPartitions(SparkSession spark, TableIdentifier tableIdent) {
+  public static List<SparkPartition> getPartitions(SparkSession spark, TableIdentifier tableIdent,
+                                                   Optional<Map<String, String>> partitionFilter) {
     try {
       SessionCatalog catalog = spark.sessionState().catalog();
       CatalogTable catalogTable = catalog.getTableMetadata(tableIdent);
 
-      Seq<CatalogTablePartition> partitions = catalog.listPartitions(tableIdent, Option.empty());
+      Option<scala.collection.immutable.Map<String, String>> partSpec =
```

Comment:
I am not smart enough to hold this whole block in my head. Can we break this down into multiple lines?

One line to make the scala Map
One line to wrap it in optional?

Reply:
Or could just do something like
```java      
Map<String, String> pkfilter = Maps.newHashMap();
      Option<scala.collection.mutable.Map<String, String>> scalaFilter =
          Option.apply(scala.collection.JavaConverters.mapAsScalaMapConverter(pkfilter).asScala());
 ```

Reply:
But you can static import JavaConverters here too to make it shorter